pkgs/servers/monitoring/cockpit/default.nix

```
@@ -0,0 +1,223 @@
+{ lib
+, stdenv
+, fetchzip
+, fetchurl
+, fetchFromGitHub
+, autoreconfHook
+, bashInteractive
+, cacert
+, coreutils
+, dbus
+, docbook_xml_dtd_43
+, docbook_xsl
+, findutils
+, gettext
+, git
+, glib
+, glibc
+, glib-networking
+, gnused
+, gnutls
+, json-glib
+, krb5
+, libssh
+, libxcrypt
+, libxslt
+, makeWrapper
+, nodejs
+, openssh
+, openssl
+, pam
+, pkg-config
+, polkit
+, python3Packages
+, ripgrep
+, runtimeShell
+, systemd
+, udev
+, xmlto
+}:
+
+let
+  pythonWithGobject = python3Packages.python.withPackages (p: with p; [
+    pygobject3
+  ]);
+in
+
+stdenv.mkDerivation rec {
+  pname = "cockpit";
+  version = "282.1";
+
+  src = fetchFromGitHub {
+    owner = "cockpit-project";
+    repo = "cockpit";
+    rev = "83d2ca1295a47c4d2c5a2904a27f8087d2641657";
+    sha256 = "sha256-8iKLZWKplOvImHHlZKfQgyqSHGnF/jYuNbOsRycoiTM=";
+    # leaveDotGit = true;
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    makeWrapper
+    docbook_xml_dtd_43
+    docbook_xsl
+    findutils
+    gettext
+    git
+    libxslt.bin
+    nodejs
+    openssl
+    pam.out
+    pkg-config
+    python3Packages.python
+    python3Packages.setuptools
```

To avoid issues with multiple different `PYTHONPATH` during build:
```suggestion
    pythonWithGobject.python
    pythonWithGobject.setuptools
```